hono/src/services/items.ts

```diff
@@ -23,8 +23,54 @@ const selectItemSchema = createSelectSchema(items, {
   id: z.string().refine((val) => isCuid(val)),
 })
 
-app.get('/', injectDB, async (c) => {
-  return c.json(await c.get('db').query.items.findMany())
+const paginationValidation = z.object({
+  page: z
+    .string()
+    .refine((val) => !isNaN(+val) && +val >= 0, {
+      message: 'Page number must be a non-negative number',
+    })
+    .optional(),
+  size: z
+    .string()
+    .refine((val) => !isNaN(+val) && +val > 0, {
+      message: 'Page size must be a positive number',
+    })
+    .optional(),
+})
+
+app.get('/', zValidator('query', paginationValidation), injectDB, async (c) => {
+  const { page, size } = c.req.query()
+
+  const pageNumber: number = page ? +page : 0
+  const pageSize: number = size ? +size : 25
+
+  const itemsData = await c
+    .get('db')
+    .select()
+    .from(items)
+    .where(eq(items.authorUsername, 'jdoe'))
+    .limit(pageSize)
+    .offset(pageNumber)
+
+  await c.get('db').execute(sql`
+      ANALYZE items;
```

Comment:
This needed to be separate from the next magic `sql` operator to avoid type errors because the query would return `{ [{Command: ANALYZE, ... } , {Command: SELECT, ... }] }`. Typescript gave errors for array indexing that could not be resolved.